pkg/core/proxy/integrations/mysql/replayer/match.go

```
@@ -354,30 +355,58 @@ func matchClosePacket(_ context.Context, _ *zap.Logger, expected, actual mysql.P
 	return matchCount
 }
 
-func matchQueryPacket(_ context.Context, _ *zap.Logger, expected, actual mysql.PacketBundle) int {
-	matchCount := 0
+func getQueryStructure(sql string) (string, error) {
+	stmt, err := sqlparser.Parse(sql)
+	if err != nil {
+		return "", err // Propagate the parsing error
+	}
+
+	var structureParts []string
+	// Walk the AST and collect the Go type of each grammatical node.
+	_ = sqlparser.Walk(func(node sqlparser.SQLNode) (bool, error) {
+		structureParts = append(structureParts, reflect.TypeOf(node).String())
+		return true, nil
+	}, stmt)
+
+	return strings.Join(structureParts, "->"), nil
+}
+
+func matchQueryPacket(_ context.Context, log *zap.Logger, expected, actual mysql.PacketBundle) bool {
 	// Match the type and return zero if the types are not equal
 	if expected.Header.Type != actual.Header.Type {
-		return 0
-	}
-	// Match the header
-	ok := matchHeader(*expected.Header.Header, *actual.Header.Header)
-	if ok {
-		matchCount += 2
+		return false
 	}
+
 	expectedMessage, _ := expected.Message.(*mysql.QueryPacket)
 	actualMessage, _ := actual.Message.(*mysql.QueryPacket)
 
-	if len(expectedMessage.Query) != len(actualMessage.Query) {
-		return 0 // If the query lengths are not equal, return zero
+	if actual.Header.Header.PayloadLength == expected.Header.Header.PayloadLength {
+		if expectedMessage.Query == actualMessage.Query {
+			return true
+		}
 	}
 
-	// Match the query for query packet
-	if expectedMessage.Query == actualMessage.Query {
-		matchCount++
+	expectedSignature, err := getQueryStructure(expectedMessage.Query)
+	if err != nil {
+		log.Error("failed to get query structure", zap.Error(err))
+		return false
 	}
-	return matchCount
+
+	actualSignature, err := getQueryStructure(actualMessage.Query)
+	if err != nil {
+		log.Error("failed to get query structure", zap.Error(err))
```

Can we return the error here ??